### ISSUE TITLE
dest not writing last N files

### ISSUE THREAD

**Issue by UltCombo** [wearefractal/vinyl-fs#51]
I'm experiencing some weird behavior when piping a readable stream built with [merge-stream](https://github.com/grncdr/merge-stream) into  `gulp.dest` -- the last 22 files are not being written to disk nor re-emitted by `gulp.dest`:

```js
mergeStream(
  // ...
)
  .pipe(through(function(file) {
    console.log('before gulp-dest', file.relative);
    this.queue(file);
  }))
  .pipe(gulp.dest(buildPath))
  .on('end', function() {
    console.log('Application successfully built!');
  })
  .pipe(through(function(file) {
    console.log('gulp.dest re-emitted', file.relative);
    this.queue(file);
  }));
```

stdout:

```
before gulp-dest
before gulp-dest css
before gulp-dest img
gulp.dest re-emitted
before gulp-dest css/custom.css
gulp.dest re-emitted css
before gulp-dest img/bgr-header.png
before gulp-dest files
before gulp-dest js
before gulp-dest media
before gulp-dest css/fonts
before gulp-dest css/sprite-retina.css
before gulp-dest css/sprite-retina.png
gulp.dest re-emitted img
before gulp-dest css/sprite.css
before gulp-dest css/sprite.png
gulp.dest re-emitted css/custom.css
before gulp-dest css/video-js.css
before gulp-dest css/vjs
gulp.dest re-emitted img/bgr-header.png
before gulp-dest files/QC_Leadership_IrwinJacobs_ThoughtsOnLeadership.mp4
gulp.dest re-emitted files
gulp.dest re-emitted js
gulp.dest re-emitted media
gulp.dest re-emitted css/fonts
gulp.dest re-emitted css/sprite-retina.css
before gulp-dest files/QC_Leadership_LeadersCauseAnEffect.mp4
gulp.dest re-emitted css/sprite-retina.png
gulp.dest re-emitted css/sprite.css
gulp.dest re-emitted css/sprite.png
gulp.dest re-emitted css/video-js.css
gulp.dest re-emitted css/vjs
gulp.dest re-emitted files/QC_Leadership_IrwinJacobs_ThoughtsOnLeadership.mp4
before gulp-dest files/QC_Leadership_PaulJacobs_ThoughtsOnLeadership.mp4
before gulp-dest files/QC_Leadership_PaulJacobs_ThoughtsOnQualcommLeadershipDimensions.mp4
gulp.dest re-emitted files/QC_Leadership_LeadersCauseAnEffect.mp4
before gulp-dest files/QC_Leadership_SteveMollenkopf_ThoughtsOnLeadership.mp4
before gulp-dest files/QC_Leadership_WhatMakesAGreatLeader.mp4
before gulp-dest files/QualcommLeadershipDimensions_2011.pdf
before gulp-dest files/VPMeeting_PaulJacobs_2011.pdf
before gulp-dest files/VPMeeting_SteveMollenkopf_2011.pdf
before gulp-dest files/app-qmaps-ico.png
before gulp-dest files/app-qpeople-ico.png
before gulp-dest files/app-qualcommappstore-ico.png
before gulp-dest files/app-quickfacts-ico.png
before gulp-dest files/app-virtual-ico.png
before gulp-dest files/example.png
before gulp-dest files/example.pptx
before gulp-dest files/thumb-agenda-2.png
before gulp-dest files/thumb-agenda-3.png
gulp.dest re-emitted files/QC_Leadership_PaulJacobs_ThoughtsOnLeadership.mp4
gulp.dest re-emitted files/QC_Leadership_PaulJacobs_ThoughtsOnQualcommLeadershipDimensions.mp4
gulp.dest re-emitted files/QC_Leadership_SteveMollenkopf_ThoughtsOnLeadership.mp4
gulp.dest re-emitted files/QC_Leadership_WhatMakesAGreatLeader.mp4
gulp.dest re-emitted files/QualcommLeadershipDimensions_2011.pdf
gulp.dest re-emitted files/VPMeeting_PaulJacobs_2011.pdf
gulp.dest re-emitted files/VPMeeting_SteveMollenkopf_2011.pdf
gulp.dest re-emitted files/app-qmaps-ico.png
gulp.dest re-emitted files/app-qpeople-ico.png
gulp.dest re-emitted files/app-qualcommappstore-ico.png
gulp.dest re-emitted files/app-quickfacts-ico.png
gulp.dest re-emitted files/app-virtual-ico.png
gulp.dest re-emitted files/example.png
gulp.dest re-emitted files/example.pptx
gulp.dest re-emitted files/thumb-agenda-2.png
gulp.dest re-emitted files/thumb-agenda-3.png
before gulp-dest files/thumb-agenda.png
before gulp-dest files/thumb-cristianoAmon.png
before gulp-dest files/thumb-derekAberle.png
before gulp-dest files/thumb-georgeDavis.png
before gulp-dest files/thumb-howardWright.png
before gulp-dest files/thumb-irwinJacobs.png
before gulp-dest files/thumb-murthyRenduchintala.png
before gulp-dest files/thumb-paulJacobs.png
before gulp-dest files/thumb-peggyJohnson.png
before gulp-dest files/thumb-robertoPadovani.png
before gulp-dest files/thumb-steveMollenkopf.png
before gulp-dest files/thumb-steveSprigg.png
before gulp-dest img/bgr-header-small.png
before gulp-dest img/loader.gif
before gulp-dest img/ph-video-welcome.jpg
before gulp-dest img/ph-video-welcome.png
gulp.dest re-emitted files/thumb-agenda.png
gulp.dest re-emitted files/thumb-cristianoAmon.png
gulp.dest re-emitted files/thumb-derekAberle.png
gulp.dest re-emitted files/thumb-georgeDavis.png
gulp.dest re-emitted files/thumb-howardWright.png
gulp.dest re-emitted files/thumb-irwinJacobs.png
gulp.dest re-emitted files/thumb-murthyRenduchintala.png
gulp.dest re-emitted files/thumb-paulJacobs.png
gulp.dest re-emitted files/thumb-peggyJohnson.png
gulp.dest re-emitted files/thumb-robertoPadovani.png
gulp.dest re-emitted files/thumb-steveMollenkopf.png
gulp.dest re-emitted files/thumb-steveSprigg.png
gulp.dest re-emitted img/bgr-header-small.png
gulp.dest re-emitted img/loader.gif
gulp.dest re-emitted img/ph-video-welcome.jpg
gulp.dest re-emitted img/ph-video-welcome.png
before gulp-dest img/splash-small.png
before gulp-dest img/splash.png
before gulp-dest js/data.js
before gulp-dest js/device-helper.js
before gulp-dest js/excanvas.js
before gulp-dest js/fastclick.js
before gulp-dest js/fbu.pie.js
before gulp-dest js/main.js
before gulp-dest js/path.min.js
before gulp-dest js/plugins.js
before gulp-dest js/qpoll.js
before gulp-dest media/PaulJacobs_A_Thousand_Times_Yes.mp3
before gulp-dest media/PaulJacobs_A_Thousand_Times_Yes.ogg
before gulp-dest media/PaulJacobs_Absolutely_Yes.mp3
before gulp-dest media/PaulJacobs_Absolutely_Yes.ogg
before gulp-dest media/PaulJacobs_Certainly_Yes.mp3
gulp.dest re-emitted img/splash-small.png
gulp.dest re-emitted img/splash.png
gulp.dest re-emitted js/data.js
gulp.dest re-emitted js/device-helper.js
gulp.dest re-emitted js/excanvas.js
gulp.dest re-emitted js/fastclick.js
gulp.dest re-emitted js/fbu.pie.js
gulp.dest re-emitted js/main.js
gulp.dest re-emitted js/path.min.js
gulp.dest re-emitted js/plugins.js
gulp.dest re-emitted js/qpoll.js
gulp.dest re-emitted media/PaulJacobs_A_Thousand_Times_Yes.mp3
gulp.dest re-emitted media/PaulJacobs_A_Thousand_Times_Yes.ogg
gulp.dest re-emitted media/PaulJacobs_Absolutely_Yes.mp3
gulp.dest re-emitted media/PaulJacobs_Absolutely_Yes.ogg
gulp.dest re-emitted media/PaulJacobs_Certainly_Yes.mp3
before gulp-dest media/PaulJacobs_Certainly_Yes.ogg
before gulp-dest media/PaulJacobs_Definitely_Yes.mp3
before gulp-dest media/PaulJacobs_Definitely_Yes.ogg
before gulp-dest media/PaulJacobs_I_Say_Yes.mp3
before gulp-dest media/PaulJacobs_I_Say_Yes.ogg
before gulp-dest media/PaulJacobs_Of_Course_Yes.mp3
before gulp-dest media/PaulJacobs_Of_Course_Yes.ogg
before gulp-dest media/PaulJacobs_Yes.mp3
before gulp-dest media/PaulJacobs_Yes.ogg
before gulp-dest media/PaulJacobs_Yes_By_All_Means.mp3
before gulp-dest media/PaulJacobs_Yes_By_All_Means.ogg
before gulp-dest media/PaulJacobs_Yes_Yes_Yes.mp3
before gulp-dest media/PaulJacobs_Yes_Yes_Yes.ogg
before gulp-dest css/fonts/qualcomm-light.eot
before gulp-dest css/fonts/qualcomm-light.svg
before gulp-dest css/fonts/qualcomm-light.ttf
gulp.dest re-emitted media/PaulJacobs_Certainly_Yes.ogg
gulp.dest re-emitted media/PaulJacobs_Definitely_Yes.mp3
gulp.dest re-emitted media/PaulJacobs_Definitely_Yes.ogg
gulp.dest re-emitted media/PaulJacobs_I_Say_Yes.mp3
gulp.dest re-emitted media/PaulJacobs_I_Say_Yes.ogg
gulp.dest re-emitted media/PaulJacobs_Of_Course_Yes.mp3
gulp.dest re-emitted media/PaulJacobs_Of_Course_Yes.ogg
gulp.dest re-emitted media/PaulJacobs_Yes.mp3
gulp.dest re-emitted media/PaulJacobs_Yes.ogg
gulp.dest re-emitted media/PaulJacobs_Yes_By_All_Means.mp3
gulp.dest re-emitted media/PaulJacobs_Yes_By_All_Means.ogg
gulp.dest re-emitted media/PaulJacobs_Yes_Yes_Yes.mp3
gulp.dest re-emitted media/PaulJacobs_Yes_Yes_Yes.ogg
gulp.dest re-emitted css/fonts/qualcomm-light.eot
gulp.dest re-emitted css/fonts/qualcomm-light.svg
gulp.dest re-emitted css/fonts/qualcomm-light.ttf
before gulp-dest css/fonts/qualcomm-light.woff
before gulp-dest css/fonts/qualcomm-regular.eot
before gulp-dest css/fonts/qualcomm-regular.svg
before gulp-dest css/fonts/qualcomm-regular.ttf
before gulp-dest css/fonts/qualcomm-regular.woff
before gulp-dest css/fonts/qualcommweb-bold-webfont.eot
before gulp-dest css/fonts/qualcommweb-bold-webfont.svg
before gulp-dest css/fonts/qualcommweb-bold-webfont.ttf
before gulp-dest css/fonts/qualcommweb-bold-webfont.woff
before gulp-dest css/fonts/qualcommweb-light-webfont.eot
before gulp-dest css/fonts/qualcommweb-light-webfont.svg
before gulp-dest css/fonts/qualcommweb-light-webfont.ttf
before gulp-dest css/fonts/qualcommweb-light-webfont.woff
before gulp-dest css/fonts/qualcommweb-regular-webfont.eot
before gulp-dest css/fonts/qualcommweb-regular-webfont.svg
before gulp-dest css/fonts/qualcommweb-regular-webfont.ttf
before gulp-dest css/fonts/qualcommweb-regular-webfont.woff
before gulp-dest css/fonts/stylesheet.css
before gulp-dest css/vjs/vjs.eot
before gulp-dest css/vjs/vjs.svg
before gulp-dest css/vjs/vjs.ttf
before gulp-dest css/vjs/vjs.woff
Application successfully built!
```

Using Node 0.10.35, gulp 3.8.10, vinyl-fs 0.3.13.

I'll prepare a test case repository after I get home.

**Comment by laurelnaiad** [comment]
I've had these types of issues when a plugin or other stream handler in the chain isn't doing events properly. This get's especially exacerbated if you merge/split streams. It's painful to debug these issues, and it makes me think gulp could really benefit from having a rigorous plugin test harness that would hammer on the plugins to find issues (typically in `end` and `error` scenarios). :)

**Comment by UltCombo** [comment]
I can workaround the issue by having a `gulp.dest()` at the end of each of the streams being merged, then it writes to disk just fine.

The weird part is that the vinyl files are being piped into `gulp.dest()` and as far as I can see, merge-stream handles the `end` event correctly and there's no error being emitted. Of course, I didn't debug this throughout yet as I'm on work hours atm.

**Comment by UltCombo** [comment]
Thanks for the input btw @laurelnaiad :smile:

**Comment by UltCombo** [comment]
Using `event-stream`'s merge seems to work correctly, so this must be an issue with merge-stream.

**Comment by UltCombo** [comment]
FWIW this was fixed in merge-stream grncdr/merge-stream#14
